simulation/halsim_lowfi/src/main/native/cpp/NTProvider_DIO.cpp

```diff
@@ -0,0 +1,37 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "NTProvider_DIO.h"
+
+#include <HAL/Ports.h>
+#include <MockData/DIOData.h>
+
+void HALSimNTProviderDIO::Initialize() {
+  InitializeDefault(HAL_GetNumDigitalChannels(),
+                    HALSIM_RegisterDIOAllCallbacks);
+}
+
+void HALSimNTProviderDIO::OnCallback(uint32_t chan,
+                                     std::shared_ptr<nt::NetworkTable> table) {
+  table->GetEntry("init?").SetBoolean(HALSIM_GetDIOInitialized(chan));
+  table->GetEntry("value").SetBoolean(HALSIM_GetDIOValue(chan));
+  table->GetEntry("pulse_length").SetDouble(HALSIM_GetDIOPulseLength(chan));
+  table->GetEntry("input?").SetBoolean(HALSIM_GetDIOIsInput(chan));
+
+  table->GetInstance().Flush();
+}
+
+void HALSimNTProviderDIO::OnInitializedChannel(
+    uint32_t chan, std::shared_ptr<nt::NetworkTable> table) {
+  table->GetEntry("value").AddListener(
+      [chan, table](const nt::EntryNotification& ev) -> void {
```

Comment:
Do we need to be capturing the table here?

Reply:
Plus, in general, we just use `=`, rather than explicitly named captures.